ports/stm32/Makefile

```
@@ -164,7 +164,7 @@ LDFLAGS += -L"$(shell dirname $(LIBSTDCPP_FILE_NAME))"
 endif
 
 # Options for mpy-cross
-MPY_CROSS_FLAGS += -march=armv7m
+MPY_CROSS_FLAGS += $(MPY_CROSS_FLAGS_MCU_$(MCU_SERIES))
```

This might be more future-proof if we make this:
```suggestion
MPY_CROSS_FLAGS += -march=$(MPY_CROSS_ARCH_$(MCU_SERIES))
```

That way if we add a new MCU and forget to add the arch lookup, we get an error.

---

Another idea could be to do this programatically instead of using a lookup table.

pseudocode:

```
if $(CC) $(CFLAGS_MCU) -dM -E - < /dev/null | grep "__ARM_ARCH_6M__" matches, then
    ARCH = armv6m
else if $(CC) $(CFLAGS_MCU) -dM -E - < /dev/null | grep "__ARM_ARCH_7EM__" matches, then
    ARCH = armv7m
else
    error "unknown architecture"
end
```

---

I've gone with the first option (and confirmed that mpy-cross does the right thing with an empty `-march=`).

Also matched this approach on SAMD.